src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/Scripts/AzureDiskEncryptionPreRequisiteSetup.ps1

```diff
@@ -63,10 +63,20 @@ $ErrorActionPreference = "Stop"
         $now = [System.DateTime]::Now;
         $oneYearFromNow = $now.AddYears(1);
         $aadClientSecret = [Guid]::NewGuid().ToString();
-        $secureAadClientSecret = ConvertTo-SecureString -String $aadClientSecret -AsPlainText -Force;
-
         Write-Host "Creating new AAD application ($aadAppName)";
-        $ADApp = New-AzureRmADApplication -DisplayName $aadAppName -HomePage $defaultHomePage -IdentifierUris $identifierUri  -StartDate $now -EndDate $oneYearFromNow -Password $secureAadClientSecret;
+
+        #New-AzureRmADApplication cmdlet expects SecureString as -Password parameter for versions >= 5.0.0. Earlier versions expect plain string.
+        $azureResourcesModule = Get-Module 'AzureRM.Resources';
+        if($azureResourcesModule.Version.Major -ge 5)
```

Comment:
@SudhakaraReddyEvuri what kind of validation has been done around this?

Reply:
@cormacpayne  validated this on both Azure Powershell 5.3.0 and Azure Powershell 4.3.1